src/multipart.rs

```diff
@@ -307,17 +307,59 @@ impl Stream for Multipart {
                 }
             };
 
-            if &boundary_bytes[..]
-                == format!("{}{}{}", constants::BOUNDARY_EXT, boundary, constants::BOUNDARY_EXT).as_bytes()
-            {
+            if &boundary_bytes[..] == format!("{}{}", constants::BOUNDARY_EXT, boundary).as_bytes() {
+                state.stage = StreamingStage::DeterminingBoundaryType;
+            } else {
+                return Poll::Ready(Some(Err(crate::Error::IncompleteStream)));
+            }
+        }
+
+        if state.stage == StreamingStage::DeterminingBoundaryType {
+            let ext_len = constants::BOUNDARY_EXT.len();
+            let next_bytes = match stream_buffer.peek_exact(ext_len) {
```

Comment:
I *think* this check is necessary (including the new `peek_exact` function) to verify that it is `--boundary--` or `--boundary[whitespace]CRLF` and not `--boundaryZZ...`, but I would love to be proven wrong.